### PR TITLE
Rename `SparkSubmitOperator`/`SparkJDBCOperator` fields' names to comply with templated fields validation

### PR DESCRIPTION
**Pull request by shahar1**
related: #36484

This PR renames `SparkSubmitOperator`'s fields' names to comply with templated fields validation.
It also optimizes derived `SparkJDBCOperator` constructor by delegating parents' attributes to `**kwargs` to avoid repetition of parameters (templated fields were among them).
Tests were added for both operators to ensure correct templating.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/apache/spark/operators/spark_jdbc.py [hunk 1]
```diff
@@ -162,6 +165,25 @@ def __init__(
         self._hook: SparkJDBCHook | None = None
         self._use_krb5ccache = use_krb5ccache
 
+    @property
+    @deprecated(
+        reason="`_keytab` is deprecated and will be removed in the future. Please use `keytab` instead.",
```
Comment (Taragolis) [inline airflow/providers/apache/spark/operators/spark_jdbc.py]: It is not strictly required to deprecate internal/private methods/attributes. We could just remove it

**Comment by eladkal** [comment]
There are conflicts :(

**Comment by shahar1** [comment]
> There are conflicts :(

Fixed; it's expected as we remove an entry in each PR from `.pre-commit-config.yaml`, and sometimes GIT can't resolve it :)

**Review by eladkal — APPROVED** [review]

File: airflow/providers/apache/spark/operators/spark_jdbc.py [hunk 2]
```diff
@@ -44,14 +44,6 @@ class SparkJDBCOperator(SparkSubmitOperator):
...
     :param spark_jars: Additional jars to upload and add to the driver and
                        executor classpath
-    :param num_executors: number of executor to run. This should be set so as to manage
-                          the number of connections made with the JDBC database
-    :param executor_cores: Number of cores per executor
-    :param executor_memory: Memory per executor (e.g. 1000M, 2G)
-    :param driver_memory: Memory allocated to the driver (e.g. 1000M, 2G)
-    :param verbose: Whether to pass the verbose flag to spark-submit for debugging
-    :param keytab: Full path to the file that contains the keytab
-    :param principal: The name of the kerberos principal used for keytab
```
Comment (eladkal) [inline airflow/providers/apache/spark/operators/spark_jdbc.py:54]: Why are these removed?

Comment (shahar1) [reply airflow/providers/apache/spark/operators/spark_jdbc.py:54]: All of these params exist already in the parent, so to avoid repetition, we could treat them in the `**kwargs` passed to `super().__init__(...)`.

Comment (eladkal) [reply airflow/providers/apache/spark/operators/spark_jdbc.py:54]: cool thanks!
